### PR TITLE
VirtualWan CustomRouting : Add new cmdlets for HubRouteTable and modify connections

### PR DESCRIPTION
**Pull request by nagula-ritvika**
## Description
Add new cmdlets to support operations on new child resource HubRouteTable of virtual hub.
Modify existing cmdlets to support operations on new property routing configuration for connection resource in VirtualWan.

Design PR Link : https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/590

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

File: src/Network/Network/Cortex/VHubRouteTable/UpdateAzureRmVHubRouteTableCommand.cs [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+                else if (ParameterSetName.Equals(CortexParameterSetNames.ByVirtualHubObject, StringComparison.OrdinalIgnoreCase))
+                {
+                    var parentResourceId = this.ParentObject.Id;
+                    var parsedParentResourceId = new ResourceIdentifier(parentResourceId);
+                    this.ResourceGroupName = parsedParentResourceId.ResourceGroupName;
+                    this.ParentResourceName = parsedParentResourceId.ResourceName;
+                }
+
+                hubRouteTableToUpdate = this.GetVHubRouteTable(this.ResourceGroupName, this.ParentResourceName, this.Name);
+                if (hubRouteTableToUpdate == null)
```
Comment (VeryEarly) [inline src/Network/Network/Cortex/VHubRouteTable/UpdateAzureRmVHubRouteTableCommand.cs:132]: if this vhub route table was not found, will it terminate and throw exceptions?

Comment (nagula-ritvika) [reply src/Network/Network/Cortex/VHubRouteTable/UpdateAzureRmVHubRouteTableCommand.cs:132]: yes, is it not expected?

File: src/Network/Network/Cortex/VHubRouteTable/NewAzureRmRoutingConfigurationCommand.cs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Management.Automation;
+    using Microsoft.Azure.Commands.Network.Models;
+    using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+    using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+
+    [Cmdlet(
+        VerbsCommon.New,
+        ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "RoutingConfguration",
```
Comment (VeryEarly) [inline src/Network/Network/Cortex/VHubRouteTable/NewAzureRmRoutingConfigurationCommand.cs:28]: typo in cmdlet name

File: src/Network/Network/Cortex/VHubRouteTable/NewAzureRmVHubRouteCommand.cs [hunk 3]
```diff
@@ -43,6 +43,12 @@ public class NewAzureRmVHubRouteCommand : NetworkBaseCmdlet
         [ValidateNotNullOrEmpty]
         public string NextHop { get; set; }
 
+        [Parameter(
```
Comment (VeryEarly) [inline src/Network/Network/Cortex/VHubRouteTable/NewAzureRmVHubRouteCommand.cs:46]: adding mandatory parameter is breaking change and not allowed in non-breaking change release

File: src/Network/Network.Test/ScenarioTests/CortexTests.ps1 [hunk 4]
```diff
@@ -854,7 +854,7 @@ function Test-CortexExpressRouteCRUD
 function Test-VHubRouteTableCRUD {
```
Comment (VeryEarly) [inline src/Network/Network.Test/ScenarioTests/CortexTests.ps1:899]: please re-record test after modification

File: src/Network/Network/Cortex/VHubRouteTable/NewAzureRmVHubRouteCommand.cs (hunk 3 above)
Comment (nagula-ritvika) [reply src/Network/Network/Cortex/VHubRouteTable/NewAzureRmVHubRouteCommand.cs:46]: This is a completely new cmdlet being exposed for the first time, i just missed that parameter in the previous iteration

File: src/Network/Network.Test/ScenarioTests/CortexTests.ps1 (hunk 4 above)
Comment (nagula-ritvika) [reply src/Network/Network.Test/ScenarioTests/CortexTests.ps1:899]: yes, currenlty recording the test.

File: src/Network/Network/Cortex/VHubRouteTable/UpdateAzureRmVHubRouteTableCommand.cs (hunk 1 above)
Comment (VeryEarly) [reply src/Network/Network/Cortex/VHubRouteTable/UpdateAzureRmVHubRouteTableCommand.cs:132]: I think line 131 will throw exception before line 132 was executed.

You might want to add a try block for line 131 and catch the exception

Comment (nagula-ritvika) [reply src/Network/Network/Cortex/VHubRouteTable/UpdateAzureRmVHubRouteTableCommand.cs:132]: handling it with try-catch in GetVHubRouteTable method itself.

**Comment by nagula-ritvika** [comment]
> Please fix CI errors

Done :)

**Comment by number213** [comment]
Closing in favor of https://github.com/Azure/azure-powershell/pull/12172

**Pull request closed without merging**